test/language/expressions/assignment/destructuring/array-rest-put-constant-no-strict.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    The assignment target should obey `const` semantics outside of strict mode.
+es6id: 12.14.5.3
+features: [const]
+flags: [noStrict]
+---*/
+
+var value = [1];
+const c = null;
+var result;
+
+result = [ ...c ] = value;
```

Comment:
Assignment to immutable strict binding throws a TypeError in non-strict (and strict) mode (8.1.1.1.5 SetMutableBinding, step 6). (`const` is always created as immutable strict binding, cf. 15.1.8 Runtime Semantics: GlobalDeclarationInstantiation, step 15.b.i.1).